src/api.rs

```diff
@@ -186,8 +186,14 @@ pub trait Introspectable {
     ///
     /// * 'paddr' - the physical address to read from
     /// * 'buf' - the data read from memory
+    /// * 'bytes_read' - the number of bytes read
     ///
-    fn read_physical(&self, _paddr: u64, _buf: &mut [u8]) -> Result<(), Box<dyn Error>> {
+    fn read_physical(
```

Comment:
I get why the buffer is passed as an argument, but why `bytes_read` as well? Wouldn't a return value be nicer and also more in line with other low level APIs that read stuff into a buffer?

Reply:
It's true that we could have returned the amount of `bytes_read` using the `Result<>` `Ok(bytes_read)`.

Although, in the case of an Error, the caller loose this information, and he doesn't know how much bytes have already been read successfully.
So if you do a big read operation, from the start of the physicial memory, you will invevitably get an error because of unmapped pages, and you get an unusable buffer after the call.

That's why i kept it there.
But with https://github.com/Wenzel/libmicrovmi/issues/175 we can explore and redesign this concept, given that std lib `Read` trait's `read()` function is returning the amount of bytes_read as an Ok Result.

Reply:
This also mimics libvmi:
https://github.com/libvmi/libvmi/blob/master/libvmi/read.c#L284

if a page is found to be NULL, a jump is taken to the [done label](https://github.com/libvmi/libvmi/blob/master/libvmi/read.c#L303), updating the `bytes_read` according while returning a failure status

Reply:
👍